extensions/resource-deployment/src/ui/resourceTypePickerDialog.ts

```diff
@@ -209,17 +217,24 @@ export class ResourceTypePickerDialog extends DialogBase {
 				if (this.toolRefreshTimestamp !== currentRefreshTimestamp) {
 					return;
 				}
+				let enableInstallButton = false;
 				const messages: string[] = [];
 				this._toolsTable.data = toolRequirements.map(toolRef => {
 					const tool = this.toolsService.getToolByName(toolRef.name)!;
-					if (!tool.isInstalled) {
+					if (!tool.isInstalled && !tool.autoInstallSupported) {
 						messages.push(localize('deploymentDialog.ToolInformation', "{0}: {1}", tool.displayName, tool.homePage));
 						if (tool.statusDescription !== undefined) {
 							console.warn(localize('deploymentDialog.DetailToolStatusDescription', "Additional status information for tool: {0}. {1}", tool.name, tool.statusDescription));
 						}
 					}
+
+					if (!tool.isInstalled && tool.autoInstallSupported) {
+						enableInstallButton = true;
+					}
 					return [tool.displayName, tool.description, tool.isInstalled ? localize('deploymentDialog.YesText', "Yes") : localize('deploymentDialog.NoText', "No"), tool.version ? tool.version.version : ''];
 				});
+
+				this._installToolButton.hidden = !enableInstallButton;
 				this._dialogObject.okButton.enabled = messages.length === 0;
```

Comment:
Make this condition ( messages.length === 0) more direct now, unless somehow after the tools get installed after not being intiially installed message array gets cleared out somehow enabling the okButton